### PR TITLE
fix: don't rewrap auth errors

### PR DESCRIPTION
**Pull request by sl-mmuradov**
Exception handler shouldn't rewrap exceptions thrown by Authentication process

[CON-612](https://bt.smartling.net/browse/CON-612)

### CODE REVIEW

File: smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                "  \"userIdentifier\": \"invalidUser\",\n" +
+                "  \"userSecret\": \"invalidSecret\"\n" +
+                "}")
+            )
+            .willReturn(unauthorized()
+                .withHeader("Content-Type", "application/json")
+                // language=JSON
+                .withBody("{\n" +
+                    "  \"response\": {\n" +
+                    "    \"code\": \"VALIDATION_ERROR\",\n" +
```
Comment (dkBrazz) [inline smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java]: @foo4u 
This is the real response from auth-api on invalid credentials
Is it possible to get [AUTHENTICATION_ERROR](https://github.com/Smartling/java-api-sdk/blob/f00b29d17c60157375df4f06c629f020d345e0f3/smartling-api-commons/src/main/java/com/smartling/api/v2/response/ResponseCode.java#L8) here from auth-api?

Comment (foo4u) [reply smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java]: It the user provided incorrect credentials, isn't a validation error?

Comment (bloy-smartling) [reply smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java]: Changing error types is potentially an API breaking change for consumers due to a change in the contract of the response.  I would be very careful with this type of suggestion.

Comment (foo4u) [reply smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java]: I'm with Ben on not changing this for 2 reasons:

1. It breaks the current API contract (most importantly)
2. It's not what we intended `AUTHENTICATION_ERROR` for. Providing bad input to any of our APIs, even the auth API, is a `VALIDATION_ERROR`.

Comment (dkBrazz) [reply smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java]: I agree to not changing it

**Pull request merged**